code/modules/lighting/lighting_object.dm

```diff
@@ -42,7 +42,117 @@
 	affected_turf = null
 	return ..()
 
+// Look at how much we gain by moving this from the lighting object to the turf
+// If that's not enough, try using listmos, see if it helps
+// Maybe store all 4 corners as one unrolled list, would take more memory but potentially save time (not worth it, since corners are shared between turfs)
+// As an upside, this means any work we do that's the same for each lighting object can be safely cached
+// Consider precaching the max, for similar reasons (may or may not be possible, think about it (basically decreasing would be expensive potentially))
 /datum/lighting_object/proc/update()
+	switch(rand(1, 11))
```

Comment:
It would be more meaningful to have this as a sequence with a counter var to have them called equally instead of random

Reply:
I worry about falling into a rut if that makes sense, if updates happen in a "loop" you'd end up with potentially misleading profile data because one proc is getting the "easy" work
may not actually apply here but it always worries me

Reply:
Then why not make it so that they are called randomly but only once per cycle? Like have number 8 be called and then it can't be called again unless all others have been called

Reply:
honestly hadn't considered it, that'd be alright too, more exact
just rand is close enough that I hadn't thought about it, but that would be useful, something like a pick and take
thanks ghil good idea